### PR TITLE
Improve coverage for misc tasks

### PR DESCRIPTION
**Pull request by davisagli**

### CODE REVIEW

File: cumulusci/tasks/apex/testrunner.py [hunk 1]
```diff
@@ -265,19 +273,18 @@ def _get_test_results(self):
             self.logger.error("-" * 80)
             counter = 0
             for result in test_results:
-                if result["Outcome"] not in ["Fail", "CompileFail"]:
-                    continue
```
Comment (cdcarter) [inline cumulusci/tasks/apex/testrunner.py]: ... oh wow.

File: cumulusci/tasks/apex/testrunner.py [hunk 2]
```diff
@@ -308,11 +315,6 @@ def _run_task(self):
             url=self.tooling.base_url + "runTestsAsynchronous",
             json={"classids": ",".join(str(id) for id in ids)},
         )
-        if result.status_code != httplib.OK:
```
Comment (cdcarter) [inline cumulusci/tasks/apex/testrunner.py]: por que?

File: cumulusci/tasks/apex/testrunner.py (hunk 1 above)
Comment (davisagli) [reply cumulusci/tasks/apex/testrunner.py]: The logic is the same as before (I inverted the condition and moved the code that was outside it to be inside) but this got me out of adding a second test case to the fixture just to get line coverage here.

File: cumulusci/tasks/apex/testrunner.py (hunk 2 above)
Comment (davisagli) [reply cumulusci/tasks/apex/testrunner.py]: `_call_salesforce` already takes care of checking the status and raising SalesforceGeneralError

File: cumulusci/tasks/apex/testrunner.py (hunk 1 above)
Comment (cdcarter) [reply cumulusci/tasks/apex/testrunner.py]: its smart!

File: cumulusci/tasks/apex/testrunner.py [hunk 3]
```diff
@@ -374,9 +376,8 @@ def _write_output(self, test_results):
                     s += ">\n"
                     s += '    <failure type="failed" '
                     if result["Message"]:
-                        s += 'message="{}">'.format(cgi.escape(result["Message"]))
-                    else:
-                        s += ">"
```
Comment (davisagli) [inline cumulusci/tasks/apex/testrunner.py]: @cdcarter this was my other lazy coverage hack.

Once we get line coverage to 100% we can turn on branch coverage and it'll keep us honest.

Comment (cdcarter) [reply cumulusci/tasks/apex/testrunner.py]: i mean, this is also more aesthetically pleasing

Comment (davisagli) [reply cumulusci/tasks/apex/testrunner.py]: First time I've heard that phrase used to describe code that generates xml :)

**Pull request merged**